src/smt/set_defaults.cpp

```diff
@@ -192,12 +201,26 @@ void SetDefaults::setDefaultsPre(Options& opts)
       || opts.smt.proofMode == options::ProofMode::FULL
       || opts.smt.proofMode == options::ProofMode::FULL_STRICT)
   {
+    std::stringstream reasonNoProofs;
+    if (incompatibleWithProofs(opts, reasonNoProofs))
+    {
+      std::stringstream ss;
+      ss << reasonNoProofs.str() << " not supported with proofs or unsat cores";
+      throw OptionException(ss.str());
+    }
     SET_AND_NOTIFY(smt, produceProofs, true, "option requiring proofs");
   }
 
   // this check assumes the user has requested *full* proofs
   if (opts.smt.produceProofs)
   {
+    if (opts.prop.satSolver == options::SatSolverMode::CADICAL)
```

Comment:
Shouldn't this check be subsumed by the one in `incompatibleWithProofs`?

Reply:
It is not, for some reason.